### PR TITLE
Update many-count GraphQL API

### PR DESCRIPTION
**Pull request by timleslie**
This change mirrors the changes made to the top level meta queries.

### CODE REVIEW

File: packages-next/fields/src/types/relationship/Implementation.ts [hunk 1]
```diff
@@ -91,7 +91,12 @@ export class Relationship<P extends string> extends Implementation<P> {
       const filterArgs = refList.getGraphqlFilterFragment().join('\n');
       return [
         `${this.path}(${filterArgs}): [${refList.gqlNames.outputTypeName}!]!`,
-        this.withMeta ? `_${this.path}Meta(${filterArgs}): _QueryMeta` : '',
+        this.withMeta
+          ? `_${this.path}Meta(${filterArgs}): _QueryMeta @deprecated(reason: "This query will be removed in a future version. Please use ${this.path}Count instead.")`
+          : '',
+        this.withMeta
+          ? `${this.path}Count(${`where: ${refList.gqlNames.whereInputName}! = {}`}): Int!`
```
Comment (mitchellhamilton) [inline packages-next/fields/src/types/relationship/Implementation.ts:98]: This needs to be nullable (and the root one should be as well if it isn't currently)

Comment (mitchellhamilton) [inline packages-next/fields/src/types/relationship/Implementation.ts:98]: This needs to be nullable (and the root one should be as well if it isn't currently)

**Review by emmatown — CHANGES_REQUESTED** [review]

File: .changeset/few-readers-admire.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+'@keystone-next/example-todo': patch
+'@keystone-next/example-with-auth': patch
+---
+
+The GraphQL field `_all<path>Meta { count }` generated for `many` relationships has been deprecated in favour of a new field `<path>Count`, which directly returns the count.
+
+A `posts` relationship field would have the following field added to the API:
+
+```graphql
+postsCount(where: PostWhereInput! = {}): Int!
```
Comment (mitchellhamilton) [inline .changeset/few-readers-admire.md:19]:
```suggestion
postsCount(where: PostWhereInput! = {}): Int
```

**Review by emmatown — APPROVED** [review]

File: .changeset/few-readers-admire.md (hunk 2 above)
Comment (mitchellhamilton) [inline .changeset/few-readers-admire.md:19]:
```suggestion
postsCount(where: PostWhereInput! = {}): Int
```

**Pull request merged**